packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue

```diff
@@ -142,6 +177,78 @@ export default defineComponent({
   },
   setup(props) {
     const { $gettext } = useGettext()
+    const visibleItems = ref<BreadcrumbItem[]>([])
+    const hiddenItems = ref<BreadcrumbItem[]>([])
+    const displayItems = ref<BreadcrumbItem[]>(props.items.slice())
+
+    const getBreadcrumbElement = (id): HTMLElement => {
+      return document.querySelector(`[data-id="${id}"]`)
+    }
+
+    const calculateTotalBreadcrumbWidth = () => {
+      let totalBreadcrumbWidth = 0
+      visibleItems.value.forEach((item, index) => {
+        const breadcrumbElement = getBreadcrumbElement(item.id)
+        const itemClientWidth = breadcrumbElement?.getBoundingClientRect()?.width || 0
+        totalBreadcrumbWidth += itemClientWidth
+      })
+      return totalBreadcrumbWidth + 100 // 100px margin to the right to avoid breadcrumb from getting too close to the controls
+    }
+
+    const reduceBreadcrumb = (offsetIndex) => {
+      const breadcrumbMaxWidth = props.maxWidth
+      document.getElementById(props.id)?.style.setProperty('--max-width', `${breadcrumbMaxWidth}px`)
+      const totalBreadcrumbWidth = calculateTotalBreadcrumbWidth()
+
+      const isOverflowing = breadcrumbMaxWidth < totalBreadcrumbWidth
+      if (!isOverflowing || visibleItems.value.length <= 3) {
+        return
+      }
+      // Remove from the left side
+      const removed = visibleItems.value.splice(offsetIndex, 1)
+
+      hiddenItems.value.push(removed[0])
+      reduceBreadcrumb(offsetIndex)
+    }
+
+    const lastHiddenItem = computed(() => {
+      if (hiddenItems.value.length >= 1) {
+        return unref(hiddenItems)[unref(hiddenItems).length - 1]
+      }
+      return { to: {} }
+    })
+
+    const renderBreadcrumb = () => {
+      displayItems.value = props.items.slice()
+      if (displayItems.value.length > 1) {
+        displayItems.value.splice(1, 0, {
+          text: '...',
+          allowContextActions: false,
+          to: {},
+          isPreviousHiddenFolder: true
+        })
+      }
+      visibleItems.value = displayItems.value.slice()
+      hiddenItems.value = []
+
+      nextTick(() => {
+        nextTick(() => {
+          reduceBreadcrumb(2)
+        })
+      })
+    }
+
+    watch(
+      () => props.maxWidth,
```

Comment:
In vue3 watcher can watch multiple props, if in composition api, if this is not possible you can add a computed and watch this